hnn_core/optimization/general_optimization.py

```diff
@@ -101,6 +108,21 @@ def __init__(
         else:
             self.obj_fun = obj_fun  # user-defined function
             self.obj_fun_name = None
+        # Initial weights for the objective function
+        if initial_params is not None:
+            # Check if initial_params is a dict
+            if not isinstance(initial_params, dict):
```

Comment:
My only other comment would be to add a unit test for this and then it's good to go!

Reply:
```suggestion
            from .externals.mne import _validate_type  # this import would be moved to top
                    _validate_type(initial_params, types=(dict,), item_name="initial_params")
```
Also instead of using `isinstance()` you can use the `check_type()` function, this is so that there is consistency in the error messages